### PR TITLE
feature(FingerprintAuthentication)

### PR DESCRIPTION
**Pull request by elloza**
Hello Mega Android Devs!

This is just a little feature to include fingerprint authentication #18 :

I have added "Fingerprint | Alphanumerical code" to the options in  Settings ->Unlock section.
This is a simple bypass with Fingerprint Authentication, that is, if the phone has a registered fingerprint, the  fingerprint authentication is added to alphanumeric unlock screen (bypassing this screen if your fingerprint is valid and hidden it if not). If the phone has no hardware, api level or fingerprint registered, this option is disabled or it will simply not appear.

screenshot_2017-02-11-19-47-35-197_mega privacy android app
screenshot_2017-02-11-21-03-58-530_mega privacy android app

I tested in my xiaomi mi5 (API 23)  and with another phone with API level 19. It should work in lower API levels.

I dont know if you have this feature in your roadmap or if it fits in your app (may with a better design), anyway I hope it helps you :)

Great work with your app!

Have a nice weekend!

PD: some files may not be required in this pull request (icons and so on).

### CODE REVIEW

**Comment by masterjedibiadler** [comment]
Hello fellows

File: app/src/main/java/mega/privacy/android/app/lollipop/ManagerActivityLollipop.java [hunk 1]
```diff
@@ -1148,6 +1153,9 @@ protected void onCreate(Bundle savedInstanceState) {
 		aC = new AccountController(this);
 		myAccountInfo = new MyAccountInfo(this);
 
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+			megaFingerprintManager = new MegaFingerprintManager(this);
+
 		File thumbDir;
 		if (getExternalCacheDir() != null){
```
Comment (finansar222) [inline app/src/main/java/mega/privacy/android/app/lollipop/ManagerActivityLollipop.java]: นำเข้า android.widget.FrameLayout ;
 นำเข้า android.widget.ImageView ;
 นำเข้า android.widget.LinearLayout ;
+ นำเข้า android.widget.ListAdapter ;
+ นำเข้า android.widget.ListView ;
 นำเข้า android.widget.ProgressBar ;
 นำเข้า android.widget.RelativeLayout ;
 นำเข้า android.widget.ScrollView ;
@@ -132,6 +134,7 @@
 นำเข้า mega.privacy.android.app.lollipop.controllers.ChatController ;
 นำเข้า mega.privacy.android.app.lollipop.controllers.ContactController ;
 นำเข้า mega.privacy.android.app.lollipop.controllers.NodeController ;
+ import  mega.privacy.android.app.lollipop.fingerprint.MegaFingerprintManager ;
 นำเข้า mega.privacy.android.app.lollipop.listeners.ContactNameListener ;
 นำเข้า mega.privacy.android.app.lollipop.listeners.FabButtonListener ;
 นำเข้า mega.privacy.android.app.lollipop.managerSections.CameraUploadFragmentLollipop ;
@@ -267,6 +270,8 @@
boolean isFabOpen = false

**Comment by k69maddog** [comment]
Thx

File: app/src/main/java/mega/privacy/android/app/lollipop/fingerprint/FingerprintAuthenticationDialogFragmentLollipop.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Do not create a new Fragment when the Activity is re-created such as orientation changes.
+        setRetainInstance(true);
+        setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Material_Light_Dialog);
+    }
```
Comment (finansar222) [inline app/src/main/java/mega/privacy/android/app/lollipop/fingerprint/FingerprintAuthenticationDialogFragmentLollipop.java]: onCreate
STYLE_NORMAL

**Comment by 1488777** [comment]
Ja

**Comment by articlecat** [comment]
Is there something blocking fingerprint authentication from being included into the MEGA Android app? This feature is available on the iOS app, but conspicuously absent from the Android app.

Related: #18, #20

**Comment by bingoxo** [comment]
any update on this ?

**Comment by javiergm1983** [comment]
> any update on this ?

Hi @bingoxo this is ready to be included in the next v5.4 of the Android app.

**Pull request closed without merging**